updateweight2.py

```diff
@@ -0,0 +1,174 @@
+import urllib2
+import os
+import urlparse
+import json
+import multiprocessing
+import sys
+
+DIFFICULTY_KEYS_EASY = 'easy'
+DIFFICULTY_KEYS_NORMAL = 'normal'
+DIFFICULTY_KEYS_HARD = 'hard'
+DIFFICULTY_KEYS_EXPERT = 'expert'
+DIFFICULTY_KEYS_MASTER = 'master'
+
+LLSIF_WIN_API_DOMAIN = 'http://a.llsif.win/'
+LLSIF_WIN_API_ENDPOINT = 'live/json/'
+
+NOTE_TYPE_RANDOM = 0
+NOTE_TYPE_NORMAL = 1
+NOTE_TYPE_EVENT = 2
+NOTE_TYPE_HOLD = 3
+NOTE_TYPE_BOMB_1 = 4
+NOTE_TYPE_BOMB_3 = 5
+NOTE_TYPE_BOMB_5 = 6
+NOTE_TYPE_BOMB_9 = 7
+NOTE_TYPE_SWING = 11
+NOTE_TYPE_SWING_EVENT = 12
+NOTE_TYPE_SWING_HOLD = 13
+
+NOTE_WEIGHT_BASE = 1.0
+NOTE_WEIGHT_HOLD_FACTOR = 1.25
+NOTE_WEIGHT_SWING_FACTOR = 0.5
+
+STATUS_SKIPPED = 0
+STATUS_SUCCESSFUL = 1
+STATUS_ERROR = 2
+STATUS_KEYBOARD_INTERRUPT = 3
+STATUS_EXIT = 4
+
+NUMBER_OF_PROCESSES = 20
+
+FILENAME_SUCCESSFUL_LOG = 'success.json'
+FILENAME_SONG_LIST_JSON = 'newsongsjson.txt'
+
+difficultyKeys = ['easy', 'normal', 'hard', 'expert', 'master']
+
+TITLE_WELCOME = '''# Welcome to LLHelper LLSIF Song Updater!
+
+* Song data based on `%s`.
+* Program originally authored by **Chazeon** and the end of Sept 2017 in NY.
+
+* We are now updating song list file: `%s`.
+* Song log will be loaded from and written to: `%s`.
+''' % (LLSIF_WIN_API_DOMAIN, FILENAME_SONG_LIST_JSON, FILENAME_SUCCESSFUL_LOG)
+
+class Note:
+    def __init__(self, noteDict):
+        self.level = noteDict['notes_level']
+        self.type = noteDict['effect']
+        self.position = noteDict['position']
+        self.attribute = noteDict['notes_attribute']
+        self.appearTime = noteDict['timing_sec']
+        self.effectValue = noteDict['effect_value']
+    def isHold(self):
+        return self.type in [NOTE_TYPE_HOLD, NOTE_TYPE_SWING_HOLD]
+    def isSwing(self):
+        return self.type in [NOTE_TYPE_SWING, NOTE_TYPE_SWING_EVENT, NOTE_TYPE_SWING_HOLD]
+    def getNoteWeightedValue(self):
+        weightValue = NOTE_WEIGHT_BASE
+        if self.isSwing():
+            weightValue *= NOTE_WEIGHT_SWING_FACTOR
+        if self.isHold():
+            weightValue *= NOTE_WEIGHT_HOLD_FACTOR
+        return weightValue
+
+def getLiveMapJsonUrl(liveId):
+    return LLSIF_WIN_API_DOMAIN + LLSIF_WIN_API_ENDPOINT + str(liveId)
+
+def getLiveMap(liveId):
+    liveJsonUrl = getLiveMapJsonUrl(liveId)
+    liveJsonFp = urllib2.urlopen(liveJsonUrl)
```

Comment:
Make sure the request header contains good UA, Accept, Accept-Language, and Accept-Encoding field. Otherwise this may be flaky (happened to my auto upgrade script.)

Also, consider using gzip encoding here as it would dramatically reduce load latency. Using 'gzip' library will easily decode the response.